### DISCUSSION TITLE
And how to make a redirect, directly from the template?

### DISCUSSION THREAD

**Discussion by esisl** [pimcore/pimcore#12136] [Q&A / Support] [answered]
And how to make a redirect, directly from the template?
$this->redirect(...); does not work :(

**Comment by NiklasBr** [comment, marked as answer]
Make sure you are using `return $this->redirect($path);` and not just `$this->redirect($path);` in the correct controller action.

If you absolutely have exhausted all other options, you can use the [meta tag to redirect](https://www.w3.org/TR/WCAG20-TECHS/H76.html): `<meta http-equiv="refresh" content="0;URL='http://thetudors.example.com/'" />` directly in the template.

**Reply by esisl** [reply]
return $this->redirect($path);

should it work in a template?

**Reply by jdreesen** [reply]
No, templates are Twig files, this is PHP code. You have to put it in a controller.

**Comment by esisl** [comment]
It's impossible to make a redirect from a template, right?

I did so, transferred the functionality to the controller.

Sometimes (especially when modifying old code) it is convenient to redirect directly in the template:(

**Reply by jdreesen** [reply]
Like @NiklasBr said, in a template you can use `<meta http-equiv="refresh" content="0;URL='http://thetudors.example.com/'" />` but I wouldn't recommend it (only if nothing else is possible, because it's a client-side redirect).

**Comment by esisl** [comment]
Yes, I understand

Thanks
You've made a great system :)
